src/main/webapp/scripts/viewCard.js

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const MAP_THEME = [
+  {
+    featureType: 'poi.business',
+    stylers: [
+      {
+        visibility: 'off'
+      }
+    ]
+  },
+  {
+    featureType: 'poi.park',
+    elementType: 'labels.text',
+    stylers: [
+      {
+        visibility: 'off'
+      }
+    ]
+  }
+];
+
+/**
+ * Map declaration and initial setup
+ * At the beginning, retrieves the API_KEY from an external file
+ */
+const initMap = () => {
+  jQuery.get('../../../../API_KEY.txt', function (textString) {
+    const API_KEY = textString;
+
+    // Create the script tag, set the appropriate attributes.
+    const scriptMapTag = document.createElement('script');
+    scriptMapTag.src = `https://maps.googleapis.com/maps/api/js?key=${API_KEY}&callback=drawMap`;
+    scriptMapTag.defer = true;
+
+    document.head.appendChild(scriptMapTag);
+  });
+};
+
+/**
+ * Query salecard from servlet
+ * TODO: fetch from servlet
+ */
+const querySalecard = () => {
+  let salecard = vendor.saleCard;
+
+  const salecardTemplate = document.getElementById('salecard-template');
+  const salecardContainer = document.getElementById('salecard-container');
+
+  //Clean previously retrieved cards
+  salecardContainer.innerHTML = '';
+
+  let salecardCloned = salecardTemplate.content.cloneNode(true);
+
+  salecardCloned.getElementById('salecard-id').innerHTML = salecard.id;
+  salecardCloned.getElementById('business-name').innerHTML = salecard.businessName;
+  salecardCloned.getElementById('business-description').innerHTML = salecard.description;
+  salecardCloned.getElementById('openedFrom').innerHTML = salecard.startTime;
+  salecardCloned.getElementById('openedTo').innerHTML = salecard.endTime;
+  salecardCloned.getElementById('deliveryDistance').innerHTML = salecard.location.radius;
+  salecardCloned.getElementById('vendor-name').innerHTML = `${vendor.firstName} ${vendor.lastName}`;
+  salecardCloned.getElementById('vendor-phone').innerHTML = vendor.phoneNumber;
+  salecardCloned.getElementById('vendor-distance').innerHTML = `${salecard.location.distance}m`;
```

Comment:
> I think we shouldn't use `innerHTML` to set the content, because of cross-site scripting attacks. Instead use: `textContent`. [This is what I found about it](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#Security_considerations)